### PR TITLE
feat: add signOut method to medusa-js

### PR DESCRIPTION
**Pull request by suezzo**
**What**
Add signOut method to `medusa-js` package - #1094

**How**
Add a new method in the`AuthResource` with a new `StoreDeleteAuthRes` type in the `medusa` package

**Testing**
Not applicable

### CODE REVIEW

File: packages/medusa-js/src/resources/auth.ts [hunk 1]
```diff
@@ -17,6 +18,15 @@ class AuthResource extends BaseResource {
     const path = `/store/auth`
     return this.client.request("POST", path, payload, {}, customHeaders)
   }
+  
+  /**
+   * @description Removes authentication session
+   * @return {ResponsePromise<StoreDeleteAuthRes>}
```
Comment (suezzo) [inline packages/medusa-js/src/resources/auth.ts:24]: Is this empty object needed in the response? What do you think about removing it completely?

Comment (olivermrbl) [reply packages/medusa-js/src/resources/auth.ts:24]: Would `ResponsePromise<void>` be more appropriate?

File: packages/medusa-js/src/resources/auth.ts [hunk 2]
```diff
@@ -17,6 +18,15 @@ class AuthResource extends BaseResource {
...
+   */
+   signOut(): ResponsePromise<StoreDeleteAuthRes> {
```
Comment (olivermrbl) [inline packages/medusa-js/src/resources/auth.ts:26]: Rename to `deleteSession` to follow general convention

**Review by olivermrbl — CHANGES_REQUESTED** [review]

**Comment by olivermrbl** [comment]
@suezzo Would be great if you could have a look at the requested changes when time allows for it :)

**Pull request closed without merging**